js/Heap.js

```diff
@@ -12,6 +13,60 @@ const track = (event, payload) => {
   RNHeap.track(event, flatten(payload));
 };
 
+getActiveRouteName = (navigationState) => {
+  if (!navigationState) {
+    return null;
+  }
+  const route = navigationState.routes[navigationState.index];
+  // dive into nested navigators
+  if (route.routes) {
+    return `${route.routeName}::${getActiveRouteName(route)}`;
+  }
+
+  return route.routeName;
+}
+
+let navRef = null;
+const NavigationService = {
+  setTopLevelNavigator: (ref) => {
+    navRef = ref;
+  }
+}
+
+const withHeapNavigationAutotrack = (AppContainer) => {
+  return class extends React.Component {
+    componentDidMount() {
+      console.log(navRef);
+      const initialPageviewPath = getActiveRouteName(navRef.state.nav);
+      track('screenview', {
+        path: initialPageviewPath,
+      });
+      console.log(initialPageviewPath);
+    }
+
+    render() {
+      return (
+        <AppContainer
+          ref={navigatorRef => {
+            NavigationService.setTopLevelNavigator(navigatorRef);
```

Comment:
This is interesting.  Why do we need to do this?  Would it interact with any other configuration in the app?